osf/migrations/0085_populate_notification_subscriptions.py

```diff
@@ -0,0 +1,40 @@
+import logging
+
+from django.db import migrations
+
+from osf.models import NotificationSubscription, PreprintProvider
+from api.preprint_providers.permissions import GroupHelper
+
+logger = logging.getLogger(__file__)
+
+instances_created = []
+
+
+def populate_provider_notification_subscriptions(*args):
+    for provider in PreprintProvider.objects.all():
+        helper = GroupHelper(provider)
+        provider_admins = helper.get_group('admin').user_set.all()
+        provider_moderators = helper.get_group('moderator').user_set.all()
+        instance, created = NotificationSubscription.objects.get_or_create(_id='{provider_id}_preprints_added'.format(provider_id=provider._id),
+                                                                           event_name='preprints_added',
+                                                                           provider=provider)
+        if created:
+            instances_created.append(instance)
+        for user in provider_admins | provider_moderators:
+            # add user to subscription list but set their notification to none by default
+            instance.add_user_to_subscription(user, 'none', save=True)
+
+def revert(*args):
+    # The revert of this migration deletes all NotificationSubscription instances
+    for instance in instances_created:
```

Comment:
`instances_created` will only ever be an empty list here. It's only after running the forward migration (which would populate it) and having `migrate` finish (which would cause it to go out of scope) that reverse migrations are ran.

This should instead be something like `NotificationSubscription.objects.filter(provider__isnull=False).delete()`